include/mbedtls/ccm.h

```diff
@@ -143,8 +144,8 @@ int mbedtls_ccm_encrypt_and_tag( mbedtls_ccm_context *ctx, size_t length,
  * \param tag_len   The length of the tag in Bytes.
  *                  4, 6, 8, 10, 12, 14 or 16.
  *
- * \return          0 if successful and authenticated.
- * \return          #MBEDTLS_ERR_CCM_AUTH_FAILED if the tag does not match.
```

Comment:
Please retain this line. `MBEDTLS_ERR_CCM_AUTH_FAILED` is not a cipher-specific error code (it's a CCM error code, as opposed to `MBEDTLS_ERR_AES_xxx`, `MBEDTLS_ERR_CAMELLIA_xxx`, etc.). It's also important to mention it here since `MBEDTLS_ERR_CCM_AUTH_FAILED` is an “expected” error that occurs if the application receives bad data from the outside, whereas other errors would usually indicate a bug in the application or a hardware failure.